src/main/java/com/cognitive/nih/niddk/mccapi/data/primative/MccCodeableConcept.java

```diff
@@ -13,5 +15,48 @@
     @NotBlank
     private String text;
 
+    @JsonIgnore
+    public String getKey(String defSystem)
+    {
+        MccCoding fndCode=null;
+        if (!StringUtils.isEmpty(defSystem))
+        {
+            //Search the codings
+            for (MccCoding cd: coding)
+            {
+                if (cd.getSystem().compareTo(defSystem)==0)
+                {
+                    fndCode = cd;
+                    break;
+                }
+            }
+        }
+        if (fndCode == null)
+        {
+            //Default to first code
+            fndCode=coding[0];
```

Comment:
Is there any chance the coding array will be empty?  I don't see where it is populated.